airflow-core/src/airflow/ui/src/pages/Dag/Code/VersionCompareSelect.tsx

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { createListCollection, Flex, Select, type SelectValueChangeDetails, Text } from "@chakra-ui/react";
+import { useCallback, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router-dom";
+
+import { useDagVersionServiceGetDagVersions } from "openapi/queries";
+import type { DagVersionResponse } from "openapi/requests/types.gen";
+import Time from "src/components/Time";
+
+type VersionSelected = {
+  value: number;
+  version: DagVersionResponse;
+};
+
+type VersionCompareSelectProps = {
+  readonly excludeVersionNumber?: number;
```

Comment:
I wouldn't handle the `excludeVersionNumber`, worst case we compare the two same version, and diff viewer don't show any diff.

Because you can still reach that unexpected situation if you select a diff version, and then change the `selectedVersion` to the selected diff version afterhand, so it does not really matter to handle that case specifically.